cmake/Modules/FindMKLDNN.cmake

```diff
@@ -18,6 +18,47 @@ IF(NOT MKLDNN_FOUND)
 
   SET(IDEEP_ROOT "${PROJECT_SOURCE_DIR}/third_party/ideep")
   SET(MKLDNN_ROOT "${PROJECT_SOURCE_DIR}/third_party/ideep/mkl-dnn")
+
+  if(USE_XPU) # Build oneDNN GPU library
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+      set(DNNL_HOST_COMPILER "g++")
+      # g++ is soft linked to /usr/bin/cxx, oneDNN woule not treat it as an absolute path
```

Comment:
@EikanWang  @jgong5  Has an wrong git operation results in the old PR closed.... We need discuss in this PR.... Sorry for any inconvenience.

The issue here is build option DNNL_HOST_COMPILER cannot be properly handle when it is c++ (soft link g++ created by system).  oneDNN only accepts an `g++` string or an absolute path. But  oneDNN dont thinks soft link `/usr/bin/cxx` is an absolute path, and it will complain an error.